### PR TITLE
Enable search on /items and add queryables links

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):**

- None?

**Description:**

We had issues getting per-collection item search to work. Also, the queryables links were missing. Here's how we solved it. There might be better solutions, but maybe it helps someone until a proper solution is in place (there are some PRs in place that try to do similar things, e.g. https://github.com/stac-utils/stac-fastapi/pull/587 and #49)

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

File: stac_fastapi/pgstac/core.py [hunk 1]
```diff
@@ -377,14 +408,6 @@ async def get_search(  # noqa: C901
...
             ItemCollection containing items which match the search criteria.
         """
-        query_params = str(request.query_params)
-
-        # Kludgy fix because using factory does not allow alias for filter-lang
-        if filter_lang is None:
-            match = re.search(r"filter-lang=([a-z0-9-]+)", query_params, re.IGNORECASE)
-            if match:
-                filter_lang = match.group(1)
-
```
Comment (vincentsarago) [inline stac_fastapi/pgstac/core.py:387]: This is not needed anymore with the latest stac-fastapi (~3.0)

File: tests/api/test_api.py [hunk 2]
```diff
@@ -73,6 +73,13 @@ async def test_get_search_content_type(app_client):
     assert resp.headers["content-type"] == "application/geo+json"
 
 
+async def test_landing_links(app_client):
+    """test landing page links."""
+    landing = await app_client.get("/")
+    assert landing.status_code == 200, landing.text
+    assert "Queryables" in [link.get("title") for link in landing.json()["links"]]
```
Comment (vincentsarago) [inline tests/api/test_api.py:80]: make sure the `Queryables` links is added (from stac-fastapi core client)

File: tests/conftest.py [hunk 3]
```diff
@@ -137,7 +137,10 @@ def api_client(request, database):
     items_get_request_model = create_request_model(
         model_name="ItemCollectionUri",
         base_model=ItemCollectionUri,
-        mixins=[TokenPaginationExtension().GET],
+        mixins=[
+            TokenPaginationExtension().GET,
+            FilterExtension(client=FiltersClient()).GET,
+        ],
```
Comment (vincentsarago) [inline tests/conftest.py:143]: add the `filter` extension to the `/items` endpoint

**Review by vincentsarago — APPROVED** [review]

**Comment by vincentsarago** [comment]
@m-mohr this is ready to be merged. Let me know if the changed I have made are 👍

File: tests/api/test_api.py (hunk 2 above)
Comment (m-mohr) [reply tests/api/test_api.py:80]: I don't understand the comment in context of your comment here: https://github.com/stac-utils/stac-fastapi/pull/736#issuecomment-2262873514

**Comment by m-mohr** [comment]
I think it looks good, but disclaimer: I'm not overly deep into stac-fastapi.

File: tests/api/test_api.py (hunk 2 above)
Comment (vincentsarago) [reply tests/api/test_api.py:80]: Meant that the link for the queryables endpoint is added in the landing page response directly from the base client definition in stac-FastAPI so this test is kinda re-testing something that should be tested in stac-FastAPI repo

**Pull request merged**